src/julia.h

```diff
@@ -749,11 +749,14 @@ enum jlsmall_typeof_tags {
     jl_bitstags_first = jl_char_tag, // n.b. bool is not considered a bitstype, since it can be compared by pointer
     jl_max_tags = 64
 };
-extern jl_datatype_t *small_typeof[(jl_max_tags << 4) / sizeof(jl_datatype_t*)];
+#ifndef JL_LIBRARY_EXPORTS
+JL_DLLIMPORT
+#endif
```

Comment:
```suggestion
#else
JL_HIDDEN
#endif
```

Reply:
If you prefer, we could also split the name here, with a `#define jl_small_typeof small_typeof`, so that the high-performance internal name is slightly different from the exported name in the binary.

Reply:
I like that idea - `ijl_small_typeof` seems like a good choice of name consistent with how we handle functions

Reply:
It must be hidden or most platforms mess this up and duplicate the load (only windows implements it correctly otherwise)